### PR TITLE
[AreaDrawer] - having selector find all paths to include line and area

### LINKED ISSUE #1651: getAllSelections() doesn't return 'line' part of area plots
(so you can't change the styling of it)

### PR DESCRIPTION
**Pull request by bluong**
Fixes #1651

### CODE REVIEW

File: test/components/plots/areaPlotTests.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ describe("Plots", () => {
         var newTwoPointData = [{ foo: 2, bar: 1 }, { foo: 3, bar: 2 }];
         areaPlot.addDataset("newTwo", new Plottable.Dataset(newTwoPointData));
         var allAreas = areaPlot.getAllSelections("newTwo");
-        assert.strictEqual(allAreas.size(), 1, "all areas retrieved");
+        assert.strictEqual(allAreas.size(), 2, "all areas retrieved");
```
Comment (jtlan) [inline test/components/plots/areaPlotTests.ts]: "line and area retrieved"?

We should check that there is both a `class="line"` and `class="area"` `<path>` in the selection.

**Comment by jtlan** [comment]
[minor]

File: typings/d3/d3.d.ts [hunk 2]
```diff
@@ -778,7 +778,7 @@ declare module D3 {
 
         filter: {
             (filter: (data: any, index: number) => boolean, thisArg?: any): UpdateSelection;
-            //(filter: string): UpdateSelection;
```
Comment (jtlan) [inline typings/d3/d3.d.ts]: wat

**Comment by jtlan** [comment]
Send the update to DefinitelyTyped and we're good to go.

File: typings/d3/d3.d.ts (hunk 2 above)
Comment (bluong) [inline typings/d3/d3.d.ts]: I'm as confused as you are.

Comment (bluong) [inline typings/d3/d3.d.ts]: PR here
https://github.com/borisyankov/DefinitelyTyped/pull/3678

**Pull request merged**